akka-operator/templates/040-deployment.yaml

```diff
@@ -79,22 +79,35 @@ spec:
       labels:
         {{- include "akka-operator.selectorLabels" . | nindent 8 }}
     spec:
-      {{- with .Values.imagePullSecrets }}
-      imagePullSecrets:
-        {{- toYaml . | nindent 8 }}
-      {{- end }}
       serviceAccountName: {{ include "akka-operator.serviceAccountName" . }}
       securityContext:
         {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      {{- if ne .Values.lightbendSubscription.username "false" }}
+      imagePullSecrets:
+        - name: commercial-credentials
+      {{- else }}
+        {{- with .Values.imagePullSecrets }}
+        imagePullSecrets:
+          {{- toYaml . | nindent 8 }}
+        {{- end }}
+      {{- end }}
```

Comment:
When using the lightbend subscription pick the username and value from the helm-provided `commercial-credentials` secret. If using another user-provided repository, then create the secret manually and set the imagePullSecret yourself.